fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -685,7 +685,6 @@ private void addScanRangeLocations(Partition partition,
 
                 // for CBO
                 if (!collectedStat && replica.getRowCount() != -1) {
-                    cardinality += replica.getRowCount();
```

Comment:
is it right for legacy planner? maybe need to add a if statement to judge whether use nereids?

Reply:
It's absolutely wrong for either nereids or legacy planner